### DISCUSSION TITLE
How can i do a crossed over/under brace?

### DISCUSSION THREAD

**Discussion by szdytom** [typst/typst#835] [Q&A] [answered]
In Latex, I'll do:

```
$\rlap{\overbrace{\phantom{a+b+c+d}}^{1}}{a+\underbrace{b+c+d+e}_{2}}$
```

what i tried in typst:

```
#let rlap(content1, content2) = box({
	content1
	place(
		top + left,
		content2
	)
})

$ rlap(overbrace(#hide[a+b+c+d], "1"), a+underbrace(b+c+d+e, "2")) $
```

however, the overbrace is not placed in the correct position. i tried to define my own phantom:

```
#let phantom(body) = style(styles => {
	let sz = measure(body, styles)
	box(width: sz.width, height: sz.height)
})

$ rlap(overbrace(#phantom[a+b+c+d], "1"), a+underbrace(b+c+d+e, "2")) $
```

but that don't work either. Maybe typst need a offical impl of rlap?

**Comment by PgBiel** [comment]
Regarding overlapping, see https://github.com/typst/typst/issues/682, which suggests a workaround. I'm currently working on an in-library implementation; I should hopefully be able to PR within the next week.

Regarding `#hide` failing (if you're facing that at the latest release), see https://github.com/typst/typst/issues/795 (currently not fixed).

**Comment by szdytom** [comment, marked as answer]
do

```
#let rlap(content1, content2) = style(sty => {
	let w1 = measure(content1, sty).width
	let w2 = measure(content2, sty).width
	content1
	h(-w1)
	content2
	h(calc.max(w1, w2) - w2)
})

$ rlap(overbrace(#hide[a+b+c+d], "1"), a+underbrace(b+c+d+e, "2")) $
```

**Reply by ekstdo** [reply]
For anyone looking for a 0.13.0 compatible code snippet:

```typst
#let rlap(content1, content2) = context {
	let w1 = measure(content1).width
	let w2 = measure(content2).width
	content1
	h(-w1)
	content2
	h(calc.max(w1, w2) - w2)
}
```


Also what I've noticed, is that the code kinda breaks apart, as soon as you use it for anything more complicated. E.g. I replaced `b` with a `sum`:

```typst
$ rlap(overbrace(#hide($a+sum+c$), "1"), a+underbrace(sum+c+d+e, "2")) $
```

and the overbrace doesn't include `c` anymore

![image](https://github.com/user-attachments/assets/c066e829-4c16-42bb-83c5-9afdf0521b6b)
